PDFWriter/PDFParser.cpp

```diff
@@ -557,8 +559,8 @@ typedef BoxingBaseWithRW<LongFilePositionType> LongFilePositionTypeBox;
 
 
 void PDFParser::ExtendXrefToSize(XrefEntryInputVector& inXrefTable, ObjectIDType inXrefSize) {
-	while(inXrefTable.size() < inXrefSize) {
-		inXrefTable.push_back(XrefEntryInput());
+	if(inXrefTable.size() < inXrefSize) {
+		inXrefTable.insert(inXrefTable.end(), inXrefSize-inXrefTable.size(), scEmptyEntry);
```

Comment:
this could check if size > max xref table size and possibly return eFail

Reply:
i'm not that concerned that we overlook something more like that someone else does, what you already mentioned, not check that value.

it would mean less errors when using the lib but for a performance price, although i don't think its very costly to compare integers. i'm unsure.

Reply:
I agree. Its worh the safety. We get from placing it here. Ok ill open a new pr with a correction. Ill also see if given the check here we need the other ones....if not...then lets also gain from this that we dont have to check in multiple code locations.